pkg/controller/networkpolicy/validate.go

```diff
@@ -213,6 +214,36 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(curObj, oldObj interface{}
 	return reason, allowed
 }
 
+// validatePort validates if ports is valid
+func (a *antreaPolicyValidator) validatePort(ingress, egress []secv1alpha1.Rule) error {
+	isValid := func(rules []secv1alpha1.Rule) error {
+		for _, rule := range rules {
+			for _, port := range rule.Ports {
+				if port.EndPort == nil {
+					continue
+				}
+				if port.Port == nil {
+					return fmt.Errorf("if `endPort` is specified `port` must be specified")
+				}
+				if port.Port.Type == intstr.String {
+					return fmt.Errorf("if `port` is a string `endPort` cannot be specified")
+				}
+				if *port.EndPort <= port.Port.IntVal {
+					return fmt.Errorf("`endPort` should be greater than `port`")
```

Comment:
actually per the upstream KEP, the endPort should be greater than or equal to the port field.
so a 
```
Port: 8080
EndPort: 8080
```
is a valid case.. which essentially is equivalent to a single port.

we can choose not to populate the endPort field in the controlplane types if they are equal .. in case a port with a mask is slightly less efficient than a single port.

Reply:
Got it.